src/discussions/learners/data/api.test.jsx

```diff
@@ -0,0 +1,119 @@
+import MockAdapter from 'axios-mock-adapter';
+import { Factory } from 'rosie';
+
+import { getAuthenticatedHttpClient } from '@edx/frontend-platform/auth';
+import { initializeMockApp } from '@edx/frontend-platform/testing';
+
+import { initializeStore } from '../../../store';
+import { executeThunk } from '../../../test-utils';
+import {
+  getUserProfileApiUrl,
+  learnerPostsApiUrl,
+  learnersApiUrl,
+} from './api';
+import { fetchLearners, fetchUserPosts } from './thunks';
+
+import './__factories__';
+
+const courseId = 'course-v1:edX+DemoX+Demo_Course';
+const courseId2 = 'course-v1:edX+TestX+Test_Course2';
+const userProfileApiUrl = getUserProfileApiUrl();
+
+let axiosMock;
+let store;
+const username = 'abc123';
+const learnerCount = 3;
+
+describe('Learner api test cases', () => {
+  beforeEach(async () => {
+    initializeMockApp({
+      authenticatedUser: {
+        userId: 3,
+        username,
+        administrator: true,
+        roles: [],
+      },
+    });
+    Factory.resetAll();
+    store = initializeStore();
+    axiosMock = new MockAdapter(getAuthenticatedHttpClient());
+  });
+
+  afterEach(() => {
+    axiosMock.reset();
+  });
+
+  async function setupLearnerMockResponse(learnerCourseId, statusCode) {
+    axiosMock.onGet(learnersApiUrl(learnerCourseId))
+      .reply(() => [statusCode, Factory.build('learnersResult', {}, {
+        count: learnerCount,
+        pageSize: 3,
+      })]);
+
+    axiosMock.onGet(`${userProfileApiUrl}?username=learner-1,learner-2,learner-3`)
+      .reply(() => [statusCode, Factory.build('learnersProfile', {}, {
+        username: ['learner-1', 'learner-2', 'learner-3'],
+      }).profiles]);
+
+    await executeThunk(fetchLearners(courseId), store.dispatch, store.getState);
+    return store.getState().learners;
+  }
```

Comment:
The same method is repeated in the redux.test.jsx and selector.test.jsx, please check if it can be exported from one file.